### PR TITLE
Create CourseBlocks component to transition clients of ProtectedStatefulDiv

### PR DESCRIPTION
**Pull request by larrypo**
This refactor should have no effect on the HTML produced or on the behavior of the product. Its purpose is to create a pattern for future changes to more easily transition uses of &lt;ProtectedStatefulDiv> to use className="tutorial-block".

### Background

In https://github.com/code-dot-org/code-dot-org/pull/32106, I changed &lt;CourseBlocksHoc> to use className="tutorial-block" with &lt;ProtectedStatefulDiv>, and with a parent &lt;div className="tutorial-row"> instead of className="row". These changes, with new css, changed the styling of these course blocks to use flexbox instead of float:left, which allows the boxes to all grow as needed to accommodate longer international strings.

However, this is just one instance of use of &lt;ProtectedStatefulDiv> that should be migrated to the new pattern. This change will make it much easier for other clients to be migrated as well. Now, instead of changing many tags to use the correct class names, they will change to just use &lt;CourseBlocks> and list the course tiles they wish to display.

### Future work

After this change is submitted, I will migrate other users of &lt;ProtectedStatefulDiv> to use &lt;CourseBlocks> instead.

After all uses are migrated, I will remove the css that defines the float:left pattern.

## Links

- [jira](https://codedotorg.atlassian.net/browse/FND-43)
- [previous change](https://github.com/code-dot-org/code-dot-org/pull/32106)

## Testing story

This refactor does not change how the code behaves, so existing tests do not need to be updated. It has been verified by manual observation when run locally.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/studioHomepages/CourseBlocks.jsx [hunk 1]
```diff
@@ -162,33 +162,44 @@ class CourseBlocksCsfLegacy extends Component {
...
 
   render() {
     return (
       <div className="tutorial-row">
-        <ProtectedStatefulDiv className="tutorial-block" ref="0" />
-        <ProtectedStatefulDiv className="tutorial-block" ref="1" />
-        <ProtectedStatefulDiv className="tutorial-block" ref="2" />
-        <ProtectedStatefulDiv className="tutorial-block" ref="3" />
+        {this.props.tiles.map((tile, index) => (
+          <ProtectedStatefulDiv className="tutorial-block" ref={index} />
```
Comment (islemaster) [inline apps/src/templates/studioHomepages/CourseBlocks.jsx]: Looks like you're using [string refs](https://reactjs.org/docs/refs-and-the-dom.html#legacy-api-string-refs) which are now considered "legacy" (but not officially deprecated? :man_shrugging:)  Might be a good idea to use a [callback ref](https://reactjs.org/docs/refs-and-the-dom.html#callback-refs) instead, just to be more future-proof.

File: apps/src/templates/studioHomepages/CourseBlocks.jsx [hunk 2]
```diff
@@ -162,33 +162,44 @@ class CourseBlocksCsfLegacy extends Component {
   }
 }
 
-export class CourseBlocksHoc extends Component {
+export class CourseBlocks extends Component {
   static propTypes = {
-    isInternational: PropTypes.bool
+    tiles: PropTypes.arrayOf(PropTypes.string).isRequired
```
Comment (islemaster) [inline apps/src/templates/studioHomepages/CourseBlocks.jsx:168]: I wonder if it's worth a comment explaining that these are not just any strings, but jQuery selectors.

**Review by islemaster — APPROVED** [review]
LGTM!  As a follow-up, I'd love to understand why we're still using jQuery to relocate these tiles from some other part of the DOM, but seems okay for now.

File: apps/src/templates/studioHomepages/CourseBlocks.jsx (hunk 1 above)
Comment (larrypo) [reply apps/src/templates/studioHomepages/CourseBlocks.jsx]: Done.

File: apps/src/templates/studioHomepages/CourseBlocks.jsx (hunk 2 above)
Comment (larrypo) [reply apps/src/templates/studioHomepages/CourseBlocks.jsx:168]: Done.

**Review by islemaster — APPROVED** [review]
:+1: Cool.  Ship it!

**Pull request merged**